sky/cli.py

```diff
@@ -710,6 +710,21 @@ def _launch_with_confirm(
             confirm_shown = True
             click.confirm(prompt, default=True, abort=True, show_default=True)
 
+    # Lambda Labs does not support autostop or multiple nodes.
+    # If task.resources is None, cannot be Lambda Labs.
+    if task.resources:
+        for resource in task.resources:
+            if resource.cloud.is_same_cloud(sky.Lambda()):
+                if not down and idle_minutes_to_autostop is not None:
+                    with ux_utils.print_exception_no_traceback():
+                        raise exceptions.NotSupportedError(
+                            ('Lambda Labs does not support stopping '
+                             'instances.'))
+                elif task.num_nodes > 1:
+                    with ux_utils.print_exception_no_traceback():
+                        raise exceptions.NotSupportedError(
+                            ('Lambda Labs does not support --num-nodes > 1.'))
```

Comment:
Actually, we should block the case for multiple nodes in the `optimier._fill_in_launchable_resources` as the failover should skip lambda if the `--cloud` is not specified and `--num-nodes` is set to larger than 1.

We also have to add `lambda` cloud to the block list when we detect the autostop, so as to make it work with failover. Otherwise, the error will occur in the comment above.

Reply:
I'm a bit stuck on figuring out how to add the `lambda cloud to the blocklist when we launch with `autostop`. Any ideas?

Reply:
I added a piece of code in `optimizer._fill_in_launchable_resources` to skip Lambda Labs when `--num-nodes > 1`.

I don't know how to make the optimizer detect autostop, so I raise an error if `set_autostop` is called on a Lambda Labs instance (as suggested by @concretevitamin). I also do a preliminary check in `sky.launch` to see if we launch autostop on a Lambda Labs instance.